### ISSUE TITLE
 MongoDB `deleteMany` seems to send 2 identical read queries first before triggering the actual delete

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#15085]
As seen in https://github.com/prisma/prisma/issues/15084, a MongoDB `deleteMany` seems to send 2 identical read queries first before triggering the actual delete. One of those is probably not necessary:

```
$ ts-node ./script.ts
deleteMany
prisma:query db.User.aggregate([ { $match: { $expr: { }, }, }, { $project: { _id: 1, }, }, ])
prisma:query db.User.aggregate([ { $match: { $expr: { }, }, }, { $project: { _id: 1, }, }, ])
prisma:query db.User.deleteMany({ _id: { $in: [ ObjectId("6d795f757365725f69643030"), ], }, })
```

---
`4.3.0-dev.84`

**Comment by Jolg42** [comment]
Can reproduce the 2 identical queries
```
deleteMany
prisma:query db.User.aggregate([ { $match: { $expr: { }, }, }, { $project: { _id: 1, }, }, ])
prisma:query db.User.aggregate([ { $match: { $expr: { }, }, }, { $project: { _id: 1, }, }, ])
prisma:query db.User.deleteMany({ _id: { $in: [ ObjectId("6d795f757365725f69643030"), ], }, })
...
```
Note: I only see `db.User.deleteMany` line on a second run

**Comment by peakala** [comment]
Hello, I'm using
'@prisma/engines-version': 4.17.0-26.6b0aef69b7cdfc787f822ecd7cdc76d5f1991584
      prisma: 5.6.0
and I'm executing a deleteMany over a 500K documents set (i.e., the where clause selects 500K documents).

Prisma is performing indeed 2 reads before each deletion; moreover, it seems that it is chunking the deletion into several parts and after a couple of parts the transaction fails with:
"WriteConflict error: this operation conflicted with another operation. Please retry your operation or multi-document transaction" and I am 100% sure that I have 1 process running, so no other "explicit" operation is writing the same ids.

Is this fix already available in my version? Do I have to change version?

Thanks

**Comment by janpio** [comment]
No, this was merged in 5.4.0 and per your comment you are somehow still using the engines from `4.17.0`. You will have to completely upgrade to 5.4.0 or newer.
